x-pack/plugins/monitoring/server/alerts/base_alert.ts

```diff
@@ -274,46 +274,49 @@ export class BaseAlert {
     state: ExecutedState
   ) {
     const currentUTC = +new Date();
+    // for each cluster filter the nodes that belong to this cluster
     for (const cluster of clusters) {
       const nodes = data.filter((node) => node.clusterUuid === cluster.clusterUuid);
       if (!nodes.length) {
         continue;
       }
 
-      const firingNodeUuids = nodes
-        .filter((node) => node.shouldFire)
-        .map((node) => node.meta.nodeId || node.meta.instanceId)
-        .join(',');
-      const instanceId = `${this.alertOptions.id}:${cluster.clusterUuid}:${firingNodeUuids}`;
-      const instance = services.alertInstanceFactory(instanceId);
-      const newAlertStates: AlertNodeState[] = [];
       const key = this.alertOptions.accessorKey;
+
+      // for each node, update the alert's state with node state
```

Comment:
Here is where I've moved the triggering of the new alert into the inner for loop that loops through each node, instead of in the cluster for loop.